### ISSUE TITLE
Real Spread Radius

### ISSUE THREAD

**Issue by zfzr031** [sh4rkman/SquadCalc#216]
It is obvious that the Spread should be a circle at 100m. 
(Use 'Custom Training' mod to generate landing points)
![PixPin_2024-05-14_12-54-24](https://github.com/user-attachments/assets/2f3371a7-5aeb-4c3b-b50c-991bda21b7e8)
![PixPin_2024-11-01_11-29-41](https://github.com/user-attachments/assets/1cd81b66-48cd-450b-b544-234b9a2be56e)

**Comment by sh4rkman** [comment]
I'm going to need help on this.

If i'm pretty sure my spreads calculations are on on points at high range, i have to admit at close range there's something wrong, either the vertical spread is too high, or the horizontal is to low.

The current logic is described [in the wiki ](https://github.com/sh4rkman/SquadCalc/wiki/Deducing-Spread), if someone wants to investigate further, check if OWI have some maximum vertical spread or something.

**Comment by zfzr031** [comment]
> I'm going to need help on this.
> 
> If i'm pretty sure my spreads calculations are on on points at high range, i have to admit at close range there's something wrong, either the vertical spread is too high, or the horizontal is to low.
> 
> The current logic is described [in the wiki ](https://github.com/sh4rkman/SquadCalc/wiki/Deducing-Spread), if someone wants to investigate further, check if OWI have some maximum vertical spread or something.

When approximately perpendicular, vertical spread and horizontal spread should be equal, but the calculation result is not like this. However, I checked wiki and didn't find any logical errors, which is too strange.

By the way, the actual result at a distance of 100 meters horizontal spread is twice the calculated value, but it is equal at the maximum distance......this is confusing...$^%*&%^$

**Comment by zfzr031** [comment]
I think I have found the answer. The following is an explanation:
After abandoning the misleading concept of MOA, I suddenly realized that the vertical component of velocity and time should be used to determine the radius of impact point dispersion.
Then I realized the mistake in the Reducing Spread, as shown in the following picture. The actual MOA that should be calculated is the projection on the horizontal plane.
The above two points should explain why the farther the distance, the smaller the horizontal spread. In terms of calculation results, it can also ensure that it is equal to the vertical spread at a 90° position.
(I'm sorry I can't provide a specific process because I haven't been using trigonometric functions for many years)
![PixPin_2024-11-26_20-44-28](https://github.com/user-attachments/assets/40701e5c-7a1e-4761-8132-13a31c619876)

**Comment by sh4rkman** [comment]
I didn't understand that last message, nor the image attached but some response anyway : 

`The above two points should explain why the farther the distance, the smaller the horizontal spread.`

Are we not mixing horizontal&vertical spread ? For reference i call "Vertical" the result of applying MOA to the elevation of mortar (mils ingame) and horizontal the result of applying MOA to the bearing (where we aim the mortar) : 

![353354942-44bdfbc0-1666-4f57-ac5d-e0de83cc25b4](https://github.com/user-attachments/assets/eb0aac9e-4406-4a1d-b80b-4ad06f9d59b8)

If we talking about the same things then i disagree horizontal spread tend to big bigger at long range than low range, because the projectile path when shooting close [tend to be shorter](https://github.com/sh4rkman/SquadCalc/wiki/Deducing-Spread#horizontal-spread-1).

If you were talking about how vertical spread drastically reduce at long range then yeah i agree.

---

That being said by digging and testing again a lot in Training_Range modded map, i came to the conclusion that it's definitly the horizontal spread that is divided by 2. By literally cheating the numbers and applying a x2 on it, it's now a perfect representation at all ranges (confirmed by spamming 200 rounds with AdminSlomo 20)

This highly suggest that i have a error somewhere in code where i divide by 360 instead of 180 when manipulating angles, or something like that.

![20241126_23h25m_iCNnyNB7b4e](https://github.com/user-attachments/assets/c5c9c104-4594-4c80-8d6c-f93ec6f9b57e)

Now, even if i find the mathematical error that i have and fix it i'll have another problem : UB32 and BM21 grad spreads are pretty realist right now. The x2 fix the horizontal spread for mortar/techmortar/m121 (didn't test hell mortar) but it completly fuckup UB & Grad with unreallistically large horizontal spread.

Which means... this is either something when calculating horizontal spread with "High Angles" weapon OR the gravity scale that UB32 & Grad have (x2) have been "fixing" my wrong calcs for a long time.

In the second case that would mean that whatever the error is, horizontal spread doesn't take x2 gravity scale into account.. that's sketchy

**Comment by zfzr031** [comment]
For any initial velocity with the maximum offset at any angle, there is always the same vertical velocity component v1. This means that without changing the MOA, only time affects the size of the horizontal spread, which is consistent with the fact that the farther the distance, the smaller the horizontal spread.
![PixPin_2024-11-27_08-52-06](https://github.com/user-attachments/assets/a5f8e6c0-2d4f-4d37-8555-22692354ccdb)

**Comment by zfzr031** [comment]
MOA can only represent the angle at the time of launch. In fact, the horizontal spread path of the projectile after launch is not an ideal surface, but a sphere, which introduces a more complex Reuleaux triangle. So I suggest using the components of speed and time in the calculation, rather than continuing to be misled by MOA.

**Comment by zfzr031** [comment]
For example, can the angle θ be directly used to calculate the arc length R1? Obviously, the formula for a planar triangle is no longer applicable here.
![PixPin_2024-11-27_18-58-27](https://github.com/user-attachments/assets/93524959-d845-4b59-a8d5-29cb97714a60)

**Comment by zfzr031** [comment]
> Ok, i understand better this image!好的，我更理解这张图片了！
> 
> > For example, can the angle θ be directly used to calculate the arc length R1?例如，角度 θ 可以直接用来计算弧长 R1 吗？
> 
> I'd say yes, perpective make it look bigger but in reality these two angles are both 90° in your example no ?我会说是的，最好让它看起来更大，但实际上这两个角度在你的例子中都是 90°，不是吗？
> 
> ![image](https://private-user-images.githubusercontent.com/9033912/390424381-08b578e9-e811-4a58-b064-76689a7df6e3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MTM5ODIsIm5iZiI6MTczMjcxMzY4MiwicGF0aCI6Ii85MDMzOTEyLzM5MDQyNDM4MS0wOGI1NzhlOS1lODExLTRhNTgtYjA2NC03NjY4OWE3ZGY2ZTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjdUMTMyMTIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTZmODQzMmRiZTE2OWIxMDg2NDYzZjQ1Yzk5NTU3Mzk1NTBiZjM0Y2VlZmY3YmU2NGUwYWYxZTdiZGU1NzZmMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Tm_n1oWRfAq_IH_v5NMz9aK73wl3ay6xagFQQkhCwVE)

This is a coordinate system. θ is clearly an acute angle.

**Comment by sh4rkman** [comment]
Fuck, yeah you are right this can't be 90°, ok i'll need some hours and plenty of coffee my brain isn't ready for things like that

**Comment by zfzr031** [comment]
What I mean is, the calculation that provides the easiest answer has already been given above.

For any initial velocity with the maximum offset at any angle, there is always the same vertical velocity component v1. This means that without changing the MOA, only time affects the size of the horizontal spread, which is consistent with the fact that the farther the distance, the smaller the horizontal spread.

This simple calculation avoids a lot of trouble. Although it is an approximation, due to the small MOA, the calculated error is negligible.
![PixPin_2024-11-27_08-52-06](https://github.com/user-attachments/assets/166a8b96-2859-4321-85e0-52ddb284c327)

**Comment by zfzr031** [comment]
> What I mean is, the calculation that provides the easiest answer has already been given above.我的意思是，上面已经给出了提供最简单答案的计算。
> 
> For any initial velocity with the maximum offset at any angle, there is always the same vertical velocity component v1. This means that without changing the MOA, only time affects the size of the horizontal spread, which is consistent with the fact that the farther the distance, the smaller the horizontal spread.对于在任何角度上具有最大偏移量的任何初始速度，始终存在相同的垂直速度分量 v1。这意味着在不更改 MOA 的情况下，只有时间会影响水平传播的大小，这与距离越远，水平传播越小的事实是一致的。
> 
> This simple calculation avoids a lot of trouble. Although it is an approximation, due to the small MOA, the calculated error is negligible.这个简单的计算避免了很多麻烦。虽然这是一个近似值，但由于 MOA 较小，计算误差可以忽略不计。 ![PixPin_2024-11-27_08-52-06](https://private-user-images.githubusercontent.com/145041173/390429978-166a8b96-2859-4321-85e0-52ddb284c327.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MTUwNzUsIm5iZiI6MTczMjcxNDc3NSwicGF0aCI6Ii8xNDUwNDExNzMvMzkwNDI5OTc4LTE2NmE4Yjk2LTI4NTktNDMyMS04NWUwLTUyZGRiMjg0YzMyNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QxMzM5MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hOWE5MDdmMTc4MGY2OTk3ZjVmNWY3MTlmZTMzNTJjNzNmYjI5ZjQ1M2ZhYWY1NjRkOWNjNjcwYTY0MWE5ZmU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.PZEs_i3nzlcoacpq7-hBE25An_ctANyJvhbiG4SmgLE)

Otherwise, a strange ellipse will be drawn at a distance of 100 meters or closer, which I am sure will not be helpful for practical use.

**Comment by sh4rkman** [comment]
> For example, can the angle θ be directly used to calculate the arc length R1? Obviously, the formula for a planar triangle is no longer applicable here. ![PixPin_2024-11-27_19-04-07](https://private-user-images.githubusercontent.com/145041173/390381938-163dc621-1681-4b50-a81b-53ac0b69b795.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MTU0NTUsIm5iZiI6MTczMjcxNTE1NSwicGF0aCI6Ii8xNDUwNDExNzMvMzkwMzgxOTM4LTE2M2RjNjIxLTE2ODEtNGI1MC1hODFiLTUzYWMwYjY5Yjc5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QxMzQ1NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kN2U4MGI4OWM3ZDI1ZGE0ZGJhMjhhMDAyOGYwYjA5YmNmOGVjNzMyZDlmYTg3YzYzOTIzM2E3NDQyYjM5N2E1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wHG3E_EzHS3kM4sPAT4rbv_AN_PMNUMDaf6oR3HMkwA)


  
I'm coming back to this schema because it's the only one i understand : 
To be clear i'm not using that angle right now to deduce the arc, i'm going with : 

$` arclength= \frac{θ}{360} \times  2 \pi r `$   

Where : 
* `r` is the radius (**trajectory** distance between weapon and target)
* `θ` the arc angle (the MOA in our case)

![image](https://github.com/user-attachments/assets/977c0d4c-f3e6-496a-88ef-1f4c87cffe4f)

**Comment by zfzr031** [comment]
> > For example, can the angle θ be directly used to calculate the arc length R1? Obviously, the formula for a planar triangle is no longer applicable here. <img alt="PixPin_2024-11-27_19-04-07" width="192.4857940673828" height="21.008522033691406" src="https://private-user-images.githubusercontent.com/145041173/390381938-163dc621-1681-4b50-a81b-53ac0b69b795.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MTU0NTUsIm5iZiI6MTczMjcxNTE1NSwicGF0aCI6Ii8xNDUwNDExNzMvMzkwMzgxOTM4LTE2M2RjNjIxLTE2ODEtNGI1MC1hODFiLTUzYWMwYjY5Yjc5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QxMzQ1NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kN2U4MGI4OWM3ZDI1ZGE0ZGJhMjhhMDAyOGYwYjA5YmNmOGVjNzMyZDlmYTg3YzYzOTIzM2E3NDQyYjM5N2E1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wHG3E_EzHS3kM4sPAT4rbv_AN_PMNUMDaf6oR3HMkwA">例如，角度 θ 可以直接用来计算弧长 R1 吗？显然，平面三角形的公式在这里不再适用。[ <img alt="PixPin_2024-11-27_19-04-07" width="192.4857940673828" height="21.008522033691406" src="https://private-user-images.githubusercontent.com/145041173/390381938-163dc621-1681-4b50-a81b-53ac0b69b795.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MTU0NTUsIm5iZiI6MTczMjcxNTE1NSwicGF0aCI6Ii8xNDUwNDExNzMvMzkwMzgxOTM4LTE2M2RjNjIxLTE2ODEtNGI1MC1hODFiLTUzYWMwYjY5Yjc5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QxMzQ1NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kN2U4MGI4OWM3ZDI1ZGE0ZGJhMjhhMDAyOGYwYjA5YmNmOGVjNzMyZDlmYTg3YzYzOTIzM2E3NDQyYjM5N2E1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wHG3E_EzHS3kM4sPAT4rbv_AN_PMNUMDaf6oR3HMkwA"> ](https://private-user-images.githubusercontent.com/145041173/390381938-163dc621-1681-4b50-a81b-53ac0b69b795.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MTU0NTUsIm5iZiI6MTczMjcxNTE1NSwicGF0aCI6Ii8xNDUwNDExNzMvMzkwMzgxOTM4LTE2M2RjNjIxLTE2ODEtNGI1MC1hODFiLTUzYWMwYjY5Yjc5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QxMzQ1NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kN2U4MGI4OWM3ZDI1ZGE0ZGJhMjhhMDAyOGYwYjA5YmNmOGVjNzMyZDlmYTg3YzYzOTIzM2E3NDQyYjM5N2E1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wHG3E_EzHS3kM4sPAT4rbv_AN_PMNUMDaf6oR3HMkwA)
> 
> I'm coming back to this schema because it's the only one i understand :我回到这个模式，因为它是我唯一理解的模式： To be clear i'm not using that angle right now to deduce the arc, i'm going with :需要明确的是，我现在不是用那个角度来推断弧线，我用的是：
> 
> a r c l e n g t h = θ 360 × 2 π r
> 
> Where : 哪里：
> 
> * `r` is the radius (**trajectory** distance between weapon and target)`r` 是半径（武器和目标之间的**弹道**距离）
> * `θ` the arc angle (the MOA in our case)`θ` 弧角（在本例中为 MOA）
> 
> ![image](https://private-user-images.githubusercontent.com/9033912/390463273-977c0d4c-f3e6-496a-88ef-1f4c87cffe4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MjA3MjksIm5iZiI6MTczMjcyMDQyOSwicGF0aCI6Ii85MDMzOTEyLzM5MDQ2MzI3My05NzdjMGQ0Yy1mM2U2LTQ5NmEtODhlZi0xZjRjODdjZmZlNGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjdUMTUxMzQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTI0ZGEwNTI5NzM1YjYzMzBiMzIwMWIwY2EzMjUxNWY2ZjljZWQwOWJhZDk4OGM5MWI3Y2YwMjBmOThkZWNjYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.AU4_YTEIa1OIGvBlgh4nROJ7tRq5hzq77D1ObVHSd4A)

Okay, but one thing to understand is that the MOA we know is the acute angle between two parabolas, not the 90° angle below.

**Comment by sh4rkman** [comment]
I swear i can't unsee that BAC = DAE = MOA  
Do you agree that if i look at this shape from the sky, all i would see is :  

![image](https://github.com/user-attachments/assets/7a6959fd-069f-4e9b-b806-a3fd62727bb2)

or i am crazy ? The trajectory is a straight line on the ground, it doesn't go like this right ?

![image](https://github.com/user-attachments/assets/3fc9586c-ba76-4766-8768-68984880b974)

**Comment by zfzr031** [comment]
> I swear i can't unsee that BAC = DAE = MOA我发誓我无法忽视 BAC = DAE = MOA Do you agree that if i look at this shape from the sky, all i would see is :你同意如果我从天空中看这个形状，我只会看到：
> 
> ![image](https://private-user-images.githubusercontent.com/9033912/390482624-7a6959fd-069f-4e9b-b806-a3fd62727bb2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MjM5NTEsIm5iZiI6MTczMjcyMzY1MSwicGF0aCI6Ii85MDMzOTEyLzM5MDQ4MjYyNC03YTY5NTlmZC0wNjlmLTRlOWItYjgwNi1hM2ZkNjI3MjdiYjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjdUMTYwNzMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Mzc4ZTc4YmU0NmZiNTkyMTlmMjc2YTllMTIwYzAxOGM1ZGQyMTAxY2QxMDA2M2M1ZTNmOGI0NWI5NzgyYmFjZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.A2Sa2r4eH16mq9MFHss6mXVPy_-IsPU8lM_19dKqAuc)
> 
> or i am crazy ? The trajectory is a straight line on the ground, it doesn't go like this right ?还是我疯了？轨迹是地面上的一条直线，它不会是这样的，对吧？
> 
> ![image](https://private-user-images.githubusercontent.com/9033912/390484201-3fc9586c-ba76-4766-8768-68984880b974.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MjM5NTEsIm5iZiI6MTczMjcyMzY1MSwicGF0aCI6Ii85MDMzOTEyLzM5MDQ4NDIwMS0zZmM5NTg2Yy1iYTc2LTQ3NjYtODc2OC02ODk4NDg4MGI5NzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjdUMTYwNzMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTA5OTU5YjRmZWJiMjE3ZGZlMmEyNjAxNzJjYWRlZmQ4OWY0MmYzOTU3NDhkYWE3OGRmZTI4NjIzMDQ2ZmI2YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.BuUK6kJWLEZgH2V8s0liQum38653LsGrCxG-NSGeVWw)

This is a triangle on the surface shape of a donut, and the correct result cannot be obtained using a planar calculation formula.

**Comment by zfzr031** [comment]
OC=OD, OA=OB, CE=ED=AF=FB, CA∥EF∥DB⊥AOB, ∠DOC=MOA, ∠EOF=weapon angle, ∠AOB is what we need
![PixPin_2024-11-28_00-24-50](https://github.com/user-attachments/assets/832aaea4-92d1-4354-9d27-44932664678f)

**Comment by zfzr031** [comment]
The relationship between the angle between the parabola and MOA should be like this.
![PixPin_2024-11-28_00-40-33](https://github.com/user-attachments/assets/01fd49fc-2d86-4b33-b8fe-69a3e54df6a0)

**Comment by sh4rkman** [comment]
(thanks god your image are great and i can clearly think)

Do you agree that here AB = CD ? (It's arcs but lets say it's a straight line to simplify yeah)

Becaude if yes then we know have a 2D plan again where we know OC/OD/OE and MOA then it's over

**Comment by zfzr031** [comment]
> (thanks god your image are great and i can clearly think)
> 
> Do you agree that here AB = CD ? (It's arcs but lets say it's a straight line to simplify yeah)
> 
> Becaude if yes then we now have a 2D plan again where we know OC/OD/OE and MOA then it's over

I agree, but ABCD is a cross-section created for calculation purposes and not an approximation.

And you should now understand that no matter how ∠EOF changes, the vector OD will always have a component vector ED perpendicular to OE.

Here, vector OD can be completely replaced by velocity, so it is easy to obtain the velocity component represented by vector ED. By multiplying the velocity component with the object's idle time , it is easy to directly obtain the actual length of ED=FB.

In fact, the AB length obtained in this step is already a very accurate value, thanks to the generally small MOA.

**Comment by sh4rkman** [comment]
Alright, i gave up understanding and i'm just implementing it without fully understanding the logic (i hate it 😑)

New horizontal formula that indeed works like a charm for every weapon and every range :

```js
    /**
     * Calculates the horizontal spread for a given trajectory path length 
     * @param {number} [timeOfFlight] - Time of flight in seconds
     * @returns {number} - Length of horizontal spread in meters
     */
    getHorizontalSpread(timeOfFlight){
        const horizontalVelocity = Math.sin(this.moa) * this.velocity;
        const horizontalSpread = horizontalVelocity * timeOfFlight;
        return Math.max(0, horizontalSpread);
    }
```

**Comment by zfzr031** [comment]
> Alright, i gave up understanding and i'm just implementing it without fully understanding the logic (i hate it 😑)好吧，我放弃了理解，我只是在没有完全理解逻辑的情况下实现它（我讨厌它😑）
> 
> New horizontal formula that indeed works like a charm for every weapon and every range :新的水平公式确实像各种武器和各种射程的魅力一样有效：
> 
> ```js
>     /**
>      * Calculates the horizontal spread for a given trajectory path length 
>      * @param {number} [timeOfFlight] - Time of flight in seconds
>      * @returns {number} - Length of horizontal spread in meters
>      */
>     getHorizontalSpread(timeOfFlight){
>         const horizontalVelocity = Math.sin(this.moa) * this.velocity;
>         const horizontalSpread = horizontalVelocity * timeOfFlight;
>         return Math.max(0, horizontalSpread);
>     }
> ```

If you are unclear about any specific steps, please indicate and I will do my best to help you understand. I am looking forward to seeing the new Deducing Spread. 🫡

**Comment by sh4rkman** [comment]
fixed with 2cef3b611bf341268a7ae7cd829571a77e55022d, let me know if results looks fine to you

**Comment by zfzr031** [comment]
> fixed with [2cef3b6](https://github.com/sh4rkman/SquadCalc/commit/2cef3b611bf341268a7ae7cd829571a77e55022d), let me know if results looks fine to you

Great job, It is now precise and error free.
